Main Cleanup of Workbench release notes

sf1919 opened the pull request:
Reordered release notes to group related work, added links, spelling & grammar check, added Workspace Calculator image

**Description of work.**
The main refactoring the the Workbench release notes ahead of release of 6.2. Includes:-

- Reordering of notes where appropriate
- Adding/fixing links
- Spelling & Grammar check
- Adding an image for the Workspace Calculator

**To test:**

- remove ```build/docs/doctrees```
- build the documents again and check there are no warnings
- Check that the links work
- Check the information is still correct
- Are the images visible and clearly illustrate the new feature/improvement?
- Grammar/Spelling check


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Updated release notes

Reordered release notes to group related work, added links, spelling & grammar check, added Workspace Calculator image

sf1919 committed:
Minor fixes

Corrected the link and image link for the Workspace Calculator and reformatted "Set as global default”.

StephenSmith25 commented on docs/source/release/v6.2.0/mantidworkbench.rst:
```diff
@@ -8,52 +8,57 @@ Mantid Workbench Changes
     :align: center
 
-- Workflow diagrams in help pages are now ``.svg`` rather than ``.png``
+- **New widget and workbench plugin:** :ref:`WorkbenchWorkspaceCalculator` **, allows to perform binary operations and scaling by a floating number on workspaces;
```
Missing a word here `New widget and workbench plugin: :ref:WorkbenchWorkspaceCalculator, allows (you/the user/users)`

StephenSmith25 commented on docs/source/release/v6.2.0/mantidworkbench.rst:
```diff
@@ -8,52 +8,57 @@ Mantid Workbench Changes
-- SliceViewer uses a more visible divider between the main data view and the peaks table view.
+- The list of eligible workspaces in the `WorkspaceSelector` can now be sorted by name.
+- Added tooltips to all the widgets in the :ref:`sliceviewer`. Please contact the developers if any are missing.
+- Script editor function call tip support for Numpy 1.21.
```
I feel like just saying code completion support covers everything, but maybe not.
```suggestion
- Script editor code completion support for Numpy 1.21.
```

StephenSmith25 commented on docs/source/release/v6.2.0/mantidworkbench.rst:
```diff
@@ -8,52 +8,57 @@ Mantid Workbench Changes
+- Script editor function call tip support for Numpy 1.21.
+- The visibility of a component parameter in the Pick tab of the :ref:`InstrumentViewer` is now steered by the 'visible' attribute of a parameter in :ref:`IPF <InstrumentParameterFile>`.
+- ADS signal handlers are now synchronized in `WorkspaceSelector`. This reduces the probability of hard crash when interacting with the widget while a script manipulating a large number of workspaces is being run.
+- Added help button to the :ref:`sliceviewer`.
```
```suggestion
- Added a help button to the :ref:`sliceviewer`.
```

StephenSmith25 commented on docs/source/release/v6.2.0/mantidworkbench.rst:
```diff
@@ -8,52 +8,57 @@ Mantid Workbench Changes
+- ``Grid`` checkbox in ``Edit axis`` dialog, and ``Grids on/off`` toolbar button will now have the correct checked state when running a plot script with major grid lines.
+- Fixed a bug in colorfill plots which lead to the loss of a spectrum from the resulting image.
+- Fixed a bug where removing the plot guess line in the fit browser could lead to an exception being thrown.
+- Fixed the advanced plotting dialog incorrectly laying out, causing the options to be partially occluded.
```
```suggestion
- Fixed the advanced plotting dialog incorrect layout, which caused options to be partially occluded.
```

StephenSmith25 commented on docs/source/release/v6.2.0/mantidworkbench.rst:
```diff
@@ -8,52 +8,57 @@ Mantid Workbench Changes
+- The visibility of a component parameter in the Pick tab of the :ref:`InstrumentViewer` is now steered by the 'visible' attribute of a parameter in :ref:`IPF <InstrumentParameterFile>`.
+- ADS signal handlers are now synchronized in `WorkspaceSelector`. This reduces the probability of hard crash when interacting with the widget while a script manipulating a large number of workspaces is being run.
+- Added help button to the :ref:`sliceviewer`.
+- :ref:`sliceviewer` can toggle between different scales again without any issue.
```
Should this be a bugfix?

StephenSmith25 reviewed (commented):
Formatting looks good, I just have a couple of wording suggestions.

sf1919 committed:
Minor changes after first review

Missing words added, some notes rephrased to make them clearer to understand, Sliceviewer scales fix moved to Bugfixes

sf1919 commented:
Thanks @StephenSmith25 . I agree with all your comments so I've made the necessary changes.

StephenSmith25 reviewed (dismissed):
Thanks! LGTM :+1:

gvardany commented on docs/source/release/v6.2.0/mantidworkbench.rst:
```diff
@@ -8,52 +8,57 @@ Mantid Workbench Changes
     :align: center
 
-- Workflow diagrams in help pages are now ``.svg`` rather than ``.png``
+- **New widget and workbench plugin:** :ref:`WorkbenchWorkspaceCalculator` **, allows users to perform binary operations and scaling by a floating number on workspaces;
```
This sentence doesn't read well, could you rephrase it?

sf1919 replied:
I've had a look at the documentation for this which is similar to this release note but perhaps gives a little more detail - 'The workspace calculator allows to quickly perform binary operations on selected workspace pairs along with simultaneous scaling by arbitrary floating point numbers, provided they are different from zero.'

Maybe I should use this in the release notes instead?

gvardany replied:
Perhaps this is more concise:
`The workspace calculator allows to quickly perform binary operations on selected workspace pairs, optionally with scaling.`

gvardany commented on docs/source/release/v6.2.0/mantidworkbench.rst:
```diff
@@ -8,52 +8,57 @@ Mantid Workbench Changes
 New and Improved
 ----------------
```
Could we split new from improved? There are 2 new widgets, it's better to make them stand out.

gvardany commented on docs/source/release/v6.2.0/mantidworkbench.rst:
```diff
@@ -8,52 +8,57 @@ Mantid Workbench Changes
-- Added help button to the sliceviewer
-- SliceViewer can toggle between different scales again without any issue.
-- SliceViewer uses a more visible divider between the main data view and the peaks table view.
+- The list of eligible workspaces in the `WorkspaceSelector` can now be sorted by name.
```
This should be removed as the widget is new in this release.

sf1919 replied:
Is it? I thought the two new widgets were Superplot and the Workspace Calculator. Is the Workspace Selector the same as the Workspace Calculator? I confess I couldn't find any documentation on the Workspace Selector so wasn't sure what it was exactly.

gvardany replied:
ahh, sorry, you are right, the selector is as old as mantid

gvardany commented on docs/source/release/v6.2.0/mantidworkbench.rst:
```diff
@@ -8,52 +8,57 @@ Mantid Workbench Changes
+- Added tooltips to all the widgets in the :ref:`sliceviewer`. Please contact the developers if any are missing.
+- Script editor code completion support for Numpy 1.21.
+- The visibility of a component parameter in the Pick tab of the :ref:`InstrumentViewer` is now steered by the 'visible' attribute of a parameter in :ref:`IPF <InstrumentParameterFile>`.
+- ADS signal handlers are now synchronized in `WorkspaceSelector`. This reduces the probability of hard crash when interacting with the widget while a script manipulating a large number of workspaces is being run.
```
This as well.

gvardany replied:
please ignore this as well

gvardany commented on docs/source/release/v6.2.0/mantidworkbench.rst:
```diff
@@ -8,52 +8,57 @@ Mantid Workbench Changes
 - Fixed the help icon not showing on OSX and high-resolution monitors.
 - Tabbing between fields in the error reporter now works as expected, rather than jumping to a random place each time.
-- Fixed the advanced plotting dialog incorrectly laying out, causing the options to be partially occluded.
+- :ref:`sliceviewer` can toggle between different scales again without any issue.
```
Can we group all the slice viewer improvements together?

gvardany reviewed (commented):
Few comments to improve the readability

sf1919 committed:
Major reordering of notes

New and improved section split into two, a new Sliceviewer section added and notes moved as required, reworded Workspace Calculator note

gvardany commented on docs/source/release/v6.2.0/mantidworkbench.rst:
```diff
@@ -5,57 +5,75 @@ Mantid Workbench Changes
+Improvements
+############
+
+- Peaks can now be added or removed from a PeaksWorkspace using the :ref:`peaks overlay <sliceviewer_peaks_overlay>` in :ref:`sliceviewer`.
```
Thanks for grouping these, however, now it seems redundant to repeat :ref:`sliceviewer` in each item as they are all under one heading. Perhaps sublists would be better. The topmost item can say `Several improvements have been done in :ref:`sliceviewer`. Then the sub-items can just enumerate them. Peaks overlay can, of course, be referenced as it is a different link.

martyngigg commented:
@gvardany This keeps attracting merge conflicts so I'd like to get this merged and we make the changes in a separate PR?
